ruby/workflows/account_transfer_workflow.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'temporalio/workflow'
+require_relative '../models/transfer_input'
+require_relative '../models/transfer_output'
+require_relative '../models/transfer_status'
+require_relative '../models/deposit_response'
+
+module Workflows
+  class AccountTransferWorkflow < Temporalio::Workflow::Definition
+    attr_reader :progress, :transfer_state, :deposit_response
+
+    def initialize
+      @progress = 0
+      @transfer_state = ''
+      @deposit_response = Models::DepositResponse.new(charge_id: '')
+    end
+
+    def execute(input)
+      workflow_type = Temporalio::Workflow.info.workflow_type
+      logger.info("Simple workflow started: #{workflow_type}, input: #{input.to_h}")
+
+      idempotency_key = Temporalio::Workflow.random.uuid
+
+      validate_transfer(input)
+      update_progress(25, 1)
+
+      withdraw_funds(idempotency_key, input.fetch('amount'), workflow_type)
+      update_progress(50, 3)
+
+      @deposit_response = deposit_funds(idempotency_key, input.fetch('amount'), workflow_type)
```

Comment:
What is the data type of `input` to `execute`, or what is the data type of `@deposit_response` here? If as I suspect they are hashes, what is the value of model objects if they are only for serialization and not deserialization? Some options are discussed at https://github.com/temporalio/sdk-ruby?tab=readme-ov-file#data-conversion.

Reply:
While fine, may consider `Struct` + `require 'json/add/struct'` for something simpler